### ISSUE TITLE
cmake_policy mess

### ISSUE THREAD

**Issue by nerdralph** [ethereum/cpp-ethereum#3229]
I think using cmake_policy is going to be a maintenance nightmare.  Having code (including build scripts) that has conditional dependencies based on different tool versions is just a bad idea.  If the build system needs functionality, just stick with the cmake_minimum_version.
For policies like CMP0028, the solution would be to fix the use of any double-colons in link dependencies.
The use of CMP0054 causes an error with cmake 3.0:
Policy "CMP0054" is not known to this version of CMake.
This is because CMP0054 was added in 3.1, and cmake_minimum_version is 3.0

**Comment by bobsummerwill** [comment]
Yeah - I am afraid I am not sufficient of a CMake expert to have much intelligent to add here other than "simpler is better".   @chfast knows more than me.

I did a pass on CMake simplification (https://github.com/ethereum/cpp-ethereum/pull/3050) and things are massively less bad that they were, but we likely have a way to go before our CMake files could be described as good.

Well - the CMakeLists.txt ones aren't too bad probably, but there is still a lot more "stuff" in:

https://github.com/ethereum/cpp-ethereum/tree/develop/cmake

... than I would really like.   Perhaps that kind of verbosity is just par-for-the-course in CMake?

**Comment by chfast** [comment]
There are no easy solutions to this problem. `cmake_minimum_version` is not enough. E.g. the CMP0054 must be set because "CMake version 3.3.2 warns when the policy is not set". The best you can get is

```
if (POLICY CMP0054)
	cmake_policy(SET CMP0054 NEW)	# No longer implicitly dereference variables.
endif()
```

We may discuss policies one by one, but not silver bullets here.

I agree about CMP0028 that it's better to fix the cmake scripts than to require old policy behavior.

**Comment by halfalicious** [comment]
@chfast : Is this still an issue or can we close this?
